java/client/src/main/java/com/youtube/vitess/client/Proto.java

```diff
@@ -203,6 +210,26 @@ public EntityId apply(Map.Entry<byte[], ?> entry) {
         // Float, Double
         this.type = Query.Type.FLOAT64;
         this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Date) {
+        // Date
+        this.type = Query.Type.DATE;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Time) {
+        // Time
+        this.type = Query.Type.TIME;
+        this.value = ByteString.copyFromUtf8(value.toString());
+      } else if (value instanceof Timestamp) {
+        // DateTime, TimeStamp
+        this.type = Query.Type.TIMESTAMP;
+        this.value = ByteString.copyFromUtf8(value.toString());
```

Comment:
I see what you're saying. The default in JDBC when you don't specify a time zone, is to use the machine's configured time zone. Since we don't offer any way to specify time zone when sending bind variables as a map, we should follow the JDBC convention of using the machine's time zone.

FYI, in #1499 I'm adding support in Cursor for getting date/time values with a given Calendar. This should make implementing ResultSet time/date methods easier. There is also a new utility class in that PR called com.youtube.vitess.mysql.DateTime, which contains format methods that will also be useful when writing PreparedStatement setDate(..., Calendar) and so on.

I think to make that work, when implementing PreparedStatement, we'll need to send date/time bind vars to the low-level Vitess client as String, after formatting them with com.youtube.vitess.mysql.DateTime.